### PR TITLE
Updated serialization to properly escape record property values

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
The record serialization was escaping property names incorrectly. This fixes it.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs [hunk 1]
```diff
@@ -233,7 +234,8 @@ public override void ToExpression(StringBuilder sb, FormulaValueSerializerSettin
 
                 flag = false;
 
-                sb.Append($"'{CharacterUtils.Escape(field.Name)}':");
+                sb.Append(IdentToken.MakeValidIdentifier(field.Name));
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs:237]: >IdentToken.MakeValidIdentifier [](http://example.com/codeflow?start=26&length=30)

Should this be called from a helper (CharacterUtils?) so we have a centralized place to handle these scenarios?

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs:237]: CharacterUtils is about escaping string values. IdentToken is the place where we deal with identifiers (which is the case for record property names). I agree it's confusing, we can consider unifying them all in a single place in the future.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs [hunk 2]
```diff
@@ -345,7 +346,8 @@ internal static void TestToString(FormulaValue result, StringBuilder sb)
                 foreach (var field in fields)
                 {
                     sb.Append(dil);
-                    sb.Append(field.Name);
+                    var escapedName = IdentToken.MakeValidIdentifier(field.Name);
+                    sb.Append(escapedName);
                     sb.Append(':');
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs:351]: ah, a bug in the test code!
Another reason to do:  #728

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs (hunk 1 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs:237]: Bonus that IdentToken.MakeValidIdentifier is public, so derived classes can call it.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**